pkg/payload/payload.go

```diff
@@ -188,10 +189,38 @@ func LoadUpdate(dir, releaseImage, excludeIdentifier string) (*Update, error) {
 	return payload, nil
 }
 
-func shouldExclude(excludeIdentifier string, manifest *lib.Manifest) bool {
-	excludeAnnotation := fmt.Sprintf("exclude.release.openshift.io/%s", excludeIdentifier)
+func shouldExclude(excludeIdentifier, profile string, manifest *lib.Manifest) bool {
 	annotations := manifest.Object().GetAnnotations()
-	return annotations != nil && annotations[excludeAnnotation] == "true"
+	if annotations == nil {
+		return false
+	}
+
+	excludeAnnotation := fmt.Sprintf("exclude.release.openshift.io/%s", excludeIdentifier)
+	if annotations[excludeAnnotation] == "true" {
+		return true
+	}
+
+	if profile == "" {
+		return false
```

Comment:
Per [this][1], we should move the empty-profile check down and have:

```go
if !shouldCheckProfile{
  return false // include manifests which declare no profiles
}

if profile == "" {
  return true // exclude manifests which declare profiles when no profile is configured
}

profileAnnotation := fmt.Sprintf("include.release.openshift.io/%s", profile)
...
```


[1]: https://github.com/openshift/cluster-version-operator/pull/404#discussion_r461551007

Reply:
done

Reply:
I just realized that with this change we can only add more manifests but not replace or deactivate default manifests. Let's say I want a different image for the ingress operator. One manifest will have the include annotation, the other will not have it. If I run the CVO with the profile, the 2 manifests will be applied.
How can we avoid this ?

Reply:
> Let's say I want a different image for the ingress operator...

Open a new PR to adjust the enhancement and say:

> Manifests with no explicit inclusion annotations implicitly belong to the `default` profile.  If no profile is configured, the cluster-version operator should use the `default` profile.

Reply:
PR opened https://github.com/openshift/enhancements/pull/414